exchange/bidder.go

```diff
@@ -87,20 +88,27 @@ type pbsOrtbSeatBid struct {
 //
 // The name refers to the "Adapter" architecture pattern, and should not be confused with a Prebid "Adapter"
 // (which is being phased out and replaced by Bidder for OpenRTB auctions)
-func adaptBidder(bidder adapters.Bidder, client *http.Client, cfg *config.Configuration, me pbsmetrics.MetricsEngine) adaptedBidder {
+func adaptBidder(bidder adapters.Bidder, client *http.Client, cfg *config.Configuration, me pbsmetrics.MetricsEngine, name openrtb_ext.BidderName) adaptedBidder {
 	return &bidderAdapter{
 		Bidder:      bidder,
+		BidderName:  name,
 		Client:      client,
 		DebugConfig: cfg.Debug,
-		me:          me,
+		me:          bidderMetrics{engine: me, disabledMetrics: cfg.Metrics.Disabled},
 	}
 }
 
 type bidderAdapter struct {
 	Bidder      adapters.Bidder
+	BidderName  openrtb_ext.BidderName
 	Client      *http.Client
 	DebugConfig config.Debug
-	me          pbsmetrics.MetricsEngine
+	me          bidderMetrics
+}
+
+type bidderMetrics struct {
+	engine          pbsmetrics.MetricsEngine
+	disabledMetrics config.DisabledMetrics
 }
 
 func (bidder *bidderAdapter) requestBid(ctx context.Context, request *openrtb.BidRequest, name openrtb_ext.BidderName, bidAdjustment float64, conversions currencies.Conversions, reqInfo *adapters.ExtraRequestInfo) (*pbsOrtbSeatBid, []error) {
```

Comment:
Since you've added `BidderName` to the `bidderAdapter`, can we please remove the `name` parameter from this function args list as it is redundant?

Reply:
`exchange/legacy.go`'s implementation of `requestBid` does use this parameter. Do you think it is better to keep it? I know it's unnecessary in every implementation of `requestBid` out there except for `exchange/legacy.go`, but for the moment, it is better to keep it to not break anything.

Another option is to eliminate `bidderAdapter`'s `BidderName` field and move `ctx = bidder.addClientTrace(ctx)` upwards from `doRequest` to `requestBid(ctx context.Context, request *openrtb.BidRequest, bidAdjustment float64, conversions currencies.Conversions, reqInfo *adapters.ExtraRequestInfo)` in order to make use of its `name` argument. Though I bet it is better not to rely on `name` given that in the future `legacy.go` will eventually be dropped and we want less arguments in function signatures. What do you think?

Reply:
Ah ok! Yeah, requestBid is part of the interface so I guess we'll have to let it be for now.